core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1582,6 +1582,42 @@ class WorkspaceService(protected val userInfo: UserInfo,
       }
     }
 
+  def retrySubmission(workspaceName: WorkspaceName, submissionRetry: SubmissionRetry, submissionId: String): Future[RetriedSubmissionReport] = {
+    getWorkspaceContextAndPermissions(workspaceName, SamWorkspaceActions.write) flatMap { workspaceContext =>
+      dataSource.inTransaction { dataAccess =>
+        withSubmission(workspaceContext, submissionId, dataAccess) { submission =>
+          val newSubmission = Submission(submissionId = submissionId,
+            submissionDate = DateTime.now(),
+            submitter = submission.submitter,
+            submissionRoot = submission.submissionRoot,
+            methodConfigurationNamespace = submission.methodConfigurationNamespace,
+            methodConfigurationName = submission.methodConfigurationName,
+            submissionEntity = submission.submissionEntity,
+            workflows = submissionRetry.retryType.filterWorkflows(submission.workflows),
+            status = submission.status,
+            useCallCache = submission.useCallCache,
+            deleteIntermediateOutputFiles = submission.deleteIntermediateOutputFiles,
+            useReferenceDisks = submission.useReferenceDisks,
+            memoryRetryMultiplier = submission.memoryRetryMultiplier,
+            workflowFailureMode = submission.workflowFailureMode,
+            externalEntityInfo = submission.externalEntityInfo,
+            userComment = submission.userComment)
```

Comment:
you can simplify this block by using `copy` (note I've also changed the submissionId):
```suggestion
          val newSubmission = submission.copy(submissionId = UUID.randomUUID().toString,
            submissionDate = DateTime.now(),
            workflows = submissionRetry.retryType.filterWorkflows(submission.workflows))
```